internal/log/logger.go

```diff
@@ -5,61 +5,68 @@
 package log
 
 import (
-	"fmt"
-	"log"
+	"github.com/influxdata/influxdb-client-go/api/log"
 )
 
-var Log Logger
+var Log LogAdapter
 
-// Logger provides filtered and categorized logging API.
-// It logs to standard logger, only errors by default
-type Logger struct {
+// LogAdapter provides filtered and categorized logging API.
+// It logs using the api logger instance. Only errors are logged by default.
+type LogAdapter struct {
 	debugLevel uint
 }
 
 // SetDebugLevel to filter log messages. Each level mean to log all categories bellow
 // 0 errors , 1 - warning, 2 - info, 3 - debug
-func (l *Logger) SetDebugLevel(debugLevel uint) {
+
+func (l *LogAdapter) SetDebugLevel(debugLevel uint) {
 	l.debugLevel = debugLevel
 }
 
-func (l *Logger) Debugf(format string, v ...interface{}) {
-	if l.debugLevel > 2 {
-		log.Print("[D]! ", fmt.Sprintf(format, v...))
+func (l *LogAdapter) Debugf(format string, v ...interface{}) {
+	if log.Log != nil && l.debugLevel > 2 {
+		log.Log.Debugf(format, v...)
 	}
 }
-func (l *Logger) Debug(msg string) {
-	if l.debugLevel > 2 {
-		log.Print("[D]! ", msg)
+
+func (l *LogAdapter) Debug(msg string) {
+	if log.Log != nil && l.debugLevel > 2 {
+		log.Log.Debug(msg)
 	}
 }
 
-func (l *Logger) Infof(format string, v ...interface{}) {
-	if l.debugLevel > 1 {
-		log.Print("[I]! ", fmt.Sprintf(format, v...))
+func (l *LogAdapter) Infof(format string, v ...interface{}) {
+	if log.Log != nil && l.debugLevel > 1 {
+		log.Log.Infof(format, v...)
 	}
 }
-func (l *Logger) Info(msg string) {
-	if l.debugLevel > 1 {
-		log.Print("[I]! ", msg)
+
+func (l *LogAdapter) Info(msg string) {
+	if log.Log != nil && l.debugLevel > 1 {
+		log.Log.Info(msg)
 	}
 }
 
-func (l *Logger) Warnf(format string, v ...interface{}) {
-	if l.debugLevel > 0 {
-		log.Print("[W]! ", fmt.Sprintf(format, v...))
+func (l *LogAdapter) Warnf(format string, v ...interface{}) {
+	if log.Log != nil && l.debugLevel > 0 {
+		log.Log.Warnf(format, v...)
 	}
 }
-func (l *Logger) Warn(msg string) {
-	if l.debugLevel > 0 {
-		log.Print("[W]! ", msg)
+
+func (l *LogAdapter) Warn(msg string) {
+	if log.Log != nil && l.debugLevel > 0 {
+		log.Log.Warn(msg)
 	}
 }
 
-func (l *Logger) Errorf(format string, v ...interface{}) {
-	log.Print("[E]! ", fmt.Sprintf(format, v...))
+func (l *LogAdapter) Errorf(format string, v ...interface{}) {
+	if log.Log != nil {
+		log.Log.Errorf(format, v...)
+	}
 }
 
-func (l *Logger) Error(msg string) {
-	log.Print("[E]! ", msg)
+func (l *LogAdapter) Error(msg string) {
+	if log.Log != nil {
+		log.Log.Error(msg)
+	}
 }
```

Comment:
There should be the default implementation with the original code.  `init()` could initially set `logger.Log`